## HW1

**CH 10**  
Factor Model  
$ \\ $  
Single Factor model  
$ R_i = E(R_i) + \beta_iF + \epsilon_i$  
Multi Factor model  
$ R_i = E[R_i] + \beta_{i,1} F_1 + \beta_{i,2}F_2 + ...+\epsilon_i $  
Seeks to improve the explanatory power by accounting for various components of systematic risk.    
$ \\ $  
Arbitrage Pricing Theory(APT)  
$ \\ $  
Arbitrage condition:  
No risk-free investment is possible.  
(because when market is efficient if arbitrage opportunity exists, people would make large volume of trades in order to make profit)  
$ \\ $  
In a single-factor security market, all well -diversified portfolio have to satisfy the expected_return - beta relationship of the CAPM to satisfy the no-arbitrage condition.  
$ \\ $  
Multifactor APT generatlizes the single-factor model to accomodate several sources of systematic risk    
$ \\ $  
Fama-French(FF) Three Factor Model  
$ R_it = \alpha_i + \beta_{iM}R_{Mt} + \beta_{iSMB}SMB_t + \beta_{iHML}HML_t +\epsilon_{it} $  
SMB - Small Minus Big, HML - High Minus Low  



## HW2

In [26]:
import pandas as pd
import numpy as np

import statsmodels.api as sm

import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

In [27]:
df = pd.read_csv("BRK13F.csv", index_col=0)
df.head()

BRK.A   MKT   SMB   HML     RF   QMJ   BAB   UMD
datadate                                                   
1/29/1999   2.01  3.63 -1.24 -3.61  0.364 -3.54 -3.87  2.34
2/26/1999  -1.53 -4.13 -3.85  2.15  0.379 -0.41  0.94 -0.40
3/31/1999   1.50  3.52 -3.85 -2.98  0.364 -2.50 -7.73 -2.51
4/30/1999   5.33  4.56  2.88  3.47  0.369 -4.54 -0.20 -9.13
5/28/1999  -2.07 -2.55  5.64  1.15  0.378  1.82  3.62 -2.98

### (1) - 3 Factor Model

$ r = R_f + \beta(R_m - R_f) + b_s \times SMB  + b_v \times HML + \alpha $  

In [28]:
returns = df['BRK.A']
rf = df['RF']
market, smb, hml = df['MKT'], df['SMB'], df['HML']

# "calculate excess return" and "market excess return"
excess_return = returns - rf
market_excess_return = market - rf
market_excess_return.name = "Rm-Rf"

In [29]:
y = excess_return
x = pd.concat([market_excess_return,smb, hml], axis=1)

In [30]:
result = sm.OLS(y, sm.add_constant(x)).fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.620
Model:                            OLS   Adj. R-squared:                  0.614
Method:                 Least Squares   F-statistic:                     95.92
Date:                Thu, 09 Mar 2023   Prob (F-statistic):           7.88e-37
Time:                        21:58:57   Log-Likelihood:                -437.21
No. Observations:                 180   AIC:                             882.4
Df Residuals:                     176   BIC:                             895.2
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.3259      0.211      1.544      0.124      -0.091       0.742
Rm-Rf          0.7444      0.047     16.002      0.000       0.653       0.836
SMB           -0.3063      0.077     -3.995      0.000      -0.458      -0.155
HML            0.5185      0.070      7.409      0.000       0.380       0.657
==============================================================================
Omnibus:                       18.985   Durbin-Watson:                   2.056
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               77.504
Skew:                           0.013   Prob(JB):                     1.48e-17
Kurtosis:                       6.215   Cond. No.                         5.10
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

$ \alpha=0.3259$($\alpha$ is not stastically significant)  
all factors in the model are significant  
Adj. R_squared = 0.614  
AIC = 882.4  
BIC = 895.2

### (2) - 4 Factor Model

$ r = R_f + \beta(R_m - R_f) + b_s \times SMB  + b_v \times HML + b_m \times UMD + \alpha $ 

In [31]:
returns = df['BRK.A']
rf = df['RF']
market, smb, hml, umd = df['MKT'], df['SMB'], df['HML'], df['UMD']

# "calculate excess return" and "market excess return"
excess_return = returns - rf
market_excess_return = market - rf
market_excess_return.name = "Rm-Rf"

In [32]:
y = excess_return
x = pd.concat([market_excess_return,smb, hml, umd], axis=1)

In [33]:
result = sm.OLS(y, sm.add_constant(x)).fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.642
Model:                            OLS   Adj. R-squared:                  0.633
Method:                 Least Squares   F-statistic:                     78.35
Date:                Thu, 09 Mar 2023   Prob (F-statistic):           5.69e-38
Time:                        22:05:45   Log-Likelihood:                -432.04
No. Observations:                 180   AIC:                             874.1
Df Residuals:                     175   BIC:                             890.0
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.3974      0.207      1.921      0.056      -0.011       0.806
Rm-Rf          0.6943      0.048     14.483      0.000       0.600       0.789
SMB           -0.3228      0.075     -4.311      0.000      -0.471      -0.175
HML            0.4879      0.069      7.088      0.000       0.352       0.624
UMD           -0.1209      0.038     -3.218      0.002      -0.195      -0.047
==============================================================================
Omnibus:                       20.214   Durbin-Watson:                   1.965
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               78.138
Skew:                           0.202   Prob(JB):                     1.08e-17
Kurtosis:                       6.202   Cond. No.                         6.49
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

$ \alpha=0.3974$($\alpha$ is not stastically significant)  
all factors in the model are significant  
Adj. R_squared = 0.633    
AIC = 874.1    
BIC = 890.0  

### (3) - 6 Factor Model

$ r = R_f + \beta(R_m - R_f) + b_s \times SMB  + b_v \times HML + b_q \times QMJ +   
  b_l \times BAB + b_m \times UMD + \alpha $ 

In [34]:
returns = df['BRK.A']
rf = df['RF']
market, smb, hml, qmj, bab, umd = df['MKT'], df['SMB'], df['HML'], df['QMJ'], df['BAB'], df['UMD']

# "calculate excess return" and "market excess return"
excess_return = returns - rf
market_excess_return = market - rf
market_excess_return.name = "Rm-Rf"

In [35]:
y = excess_return
x = pd.concat([market_excess_return,smb, hml, qmj,bab, umd], axis=1)

In [36]:
result = sm.OLS(y, sm.add_constant(x)).fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.657
Model:                            OLS   Adj. R-squared:                  0.645
Method:                 Least Squares   F-statistic:                     55.20
Date:                Thu, 09 Mar 2023   Prob (F-statistic):           1.10e-37
Time:                        22:11:16   Log-Likelihood:                -428.14
No. Observations:                 180   AIC:                             870.3
Df Residuals:                     173   BIC:                             892.6
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.2175      0.214      1.018      0.310      -0.204       0.639
Rm-Rf          0.8230      0.066     12.381      0.000       0.692       0.954
SMB           -0.2012      0.086     -2.342      0.020      -0.371      -0.032
HML            0.4726      0.080      5.942      0.000       0.316       0.630
QMJ            0.2890      0.110      2.630      0.009       0.072       0.506
BAB            0.0149      0.060      0.247      0.805      -0.104       0.134
UMD           -0.1361      0.039     -3.527      0.001      -0.212      -0.060
==============================================================================
Omnibus:                       22.582   Durbin-Watson:                   2.000
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               97.962
Skew:                           0.227   Prob(JB):                     5.34e-22
Kurtosis:                       6.585   Cond. No.                         7.94
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

$ \alpha=0.2175$($\alpha$ is not stastically significant)  
all factors in the model except "BAB" are significant  
Adj. R_squared = 0.645    
AIC = 870.3      
BIC = 892.6  

### Summary

BIC를 기준으로 볼때, 4 factor model이 가장 작은 값을 보여주었으며, 모든 factor들이 통계적으로 유의하였다. 또한 alpha의 경우에도 4-factor model에서 p-value가 가장 낮게 나타났다. 따라서 이 경우, 4-factor model이 가장 좋은 모델이라고 할 수 있으며, $\alpha$는 양수라는 것을 확인할 수 있다.  